controllers/storagecluster/routes.go

```diff
@@ -17,6 +17,9 @@ type ocsCephRGWRoutes struct{}
 // ensureCreated ensures that CephObjectStore resources exist in the desired
 // state.
 func (obj *ocsCephRGWRoutes) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	if instance.Spec.ManagedResources.CephObjectStores.DisableRoute {
```

Comment:
If I am not mistaken, It can be disabled at any point in that case we should make sure it is deleted instead of ignoring it.

Reply:
@thotz

Reply:
I consider the option was added during the installation part, hence decided to skip. With the current code if user enables this option later part, then he needs to remove the route manually. The main idea behind the fix was resetting the configuration for the route if delete it then ocs-op should recreate it, or if we add some policy to the route then it should not be reset by OCS-Op.
If you think deleting the existing route when this option is enabled if the right approach I can add that as follow up PR

Reply:
I do think that we should delete it when someone disables it later as that would be the expectation of the user. But we can ask an opinion from @jarrpa or @umangachapagain

Reply:
I am merging the PR and we can have a follow up PR because if any change has to be made then it should come to main first